bindings/matrix-sdk-ffi/src/client.rs

```diff
@@ -263,6 +270,116 @@ impl Client {
         })
     }
 
+    /// Get the URL to use to log in via Single Sign-On.
+    ///
+    /// Returns a URL that should be opened in a web browser to let the user
+    /// log in.
+    ///
+    /// After a successful login, the loginToken received at the redirect URL
+    /// should be used to log in with [`login_token`].
+    ///
+    /// # Arguments
+    ///
+    /// * `redirect_url` - The URL that will receive a `loginToken` after a
+    ///   successful SSO login.
+    ///
+    /// * `idp_id` - The optional ID of the identity provider to log in with.
+    ///
+    /// [`login_token`]: #method.login_token
+    pub async fn get_sso_login_url(
+        &self,
+        redirect_url: String,
+        idp_id: Option<String>,
+    ) -> Result<String, ClientError> {
+        Ok(self.inner.matrix_auth().get_sso_login_url(&redirect_url, idp_id.as_deref()).await?)
+    }
+
+    /// Creates a [ClientLoginBuilder] that can log into the server with a
+    /// token.
+    ///
+    /// This token is usually received in the SSO flow after following the URL
+    /// provided by [`get_sso_login_url`], note that this is not the access
+    /// token of a session.
+    ///
+    /// This should only be used for the first login.
+    ///
+    /// The [`restore_session`] method should be used to restore a logged-in
+    /// client after the first login.
+    pub fn login_token(&self, token: String) -> Result<ClientLoginBuilder, ClientError> {
+        Ok(ClientLoginBuilder::new(self.inner.matrix_auth().login_token(&token)))
+    }
+
+    /// Creates a [ClientLoginBuilder] that can log into the server by username
+    /// and password.
+    pub fn login_username(
+        &self,
+        username: String,
+        password: String,
+    ) -> Result<ClientLoginBuilder, ClientError> {
+        Ok(ClientLoginBuilder::new(self.inner.matrix_auth().login_username(username, &password)))
+    }
+
+    /// Creates a [ClientLoginBuilder] that can log into the server by email and
+    /// password.
+    pub fn login_email(
+        &self,
+        email: String,
+        password: String,
+    ) -> Result<ClientLoginBuilder, ClientError> {
+        Ok(ClientLoginBuilder::new(
+            self.inner
+                .matrix_auth()
+                .login_identifier(UserIdentifier::Email { address: email }, &password),
+        ))
+    }
+
+    /// Creates a [ClientLoginBuilder] that can log into the server by a MSISDN
+    /// phone number and password.
+    pub fn login_phone_msisdn(
```

Comment:
Instead of these three methods `login_phone_msisdn` / `login_phone` / `login_email`, could we have a `UserIdentifier` enum here at the FFI layer too, and a single method here that would take a `UserIdentifier` parameter, mimicking the SDK's API?